src/config/system_stats/battery.rs

```diff
@@ -0,0 +1,41 @@
+use anyhow::*;
+
+#[cfg(target_os = "macos")]
+use regex::Regex;
+
+pub fn get_battery_capacity() -> Result<u8> {
+    #[cfg(target_os = "linux")]
+    let capacity = linux()?;
+
+    #[cfg(target_os = "macos")]
+    let capacity = macos()?;
+
+    #[cfg(not(target_os = "macos"))]
+    #[cfg(not(target_os = "linux"))]
+    return anyhow!("Not supported OS");
+
+    Ok(capacity)
+}
+
+#[cfg(target_os = "macos")]
+fn macos() -> Result<u8> {
+    let capacity = String::from_utf8(
+        std::process::Command::new("pmset")
+            .args(&["-g", "batt"])
+            .output()
+            .context("\nError while getting the battery value on macos, with `pmset`: ")?
+            .stdout,
+    )?;
+    let regex = Regex::new(r"[0-9]*%")?;
+    let mut number = regex.captures(&capacity).unwrap().get(0).unwrap().as_str().to_string();
+    number.pop();
```

Comment:
this is,... really confusing.
what does number.pop even do here, remove a character? why? what is that character?
In this case, please explain stuff more/make code more obvious (if it's stripping whitespace, use strip/trim or whatever exists instead), and provide a comment that showcases an example of the parsed data, so it's clear what/how data looks.

Reply:
This pops of the % of the output from the command.

Reply:
yea, please do add a comment explaining that (optimally show an example of the raw data that is being parsed)